### PR TITLE
Mii Gunner small adjustments

### PR DESCRIPTION
**Pull request by SkewedAskew**
Some changes from testing, meant to have these in the original PR

Forward Tilt:
~ [-] Shoulder hitbox moved to arm

Back Air:
~ [-] Hitbox Duration: F12-15/F16-22 -> F15-18/F19-25
~ [-] FAF: 45 -> 48
~ [-] Autocancel: F38 -> F41

Charge Blast:
~ [$] Changed sfx to electric

Laser Blaze:
~ [-] The laser is now flinchless after the first 6 frames

Grenade Launch:
~ [R] Number of hits: 6 -> 4
~ [-] FAF (uncharged/max charge): 47/70 -> 52/75

Stealth Burst:
~ [$] Now has vfx during the explosion timer

Pulse Missile:
~ (*) Fixed an issue where detonation could occur on contact
~ [-] Shield Damage: 0 -> -5

Cannon Jump Kick:
~ [R] Removed spike on initial hit

### CODE REVIEW

**Comment by yoonikoo** [comment]
Hella fun. Can't stop going for Up Air spikes to suicide. LGTM.

File: fighters/miigunner/src/acmd/other.rs [hunk 1]
```diff
@@ -298,12 +321,43 @@ unsafe fn miigunner_stealthbomb_tame_effect(fighter: &mut L2CAgentBase) {
...
-		LAST_EFFECT_SET_RATE(fighter, 0.3);
-		LAST_EFFECT_SET_COLOR(fighter, 10.0, 0.15, 0.15);
-	}*/
+		EFFECT(fighter, Hash40::new("sys_smash_flash"), Hash40::new("top"), 0, 0, 0, 0, 0, 0, 0.20, 0, 0, 0, 0, 0, 0, true);
+		LAST_EFFECT_SET_COLOR(fighter, 0.5, 0.5, 5.0);
+	}
+    frame(lua_state, 100.0);
+	if is_excute(fighter) {
+		let owner_id = WorkModule::get_int(boma, *WEAPON_INSTANCE_WORK_ID_INT_LINK_OWNER) as u32;
+        if sv_battle_object::kind(owner_id) == *FIGHTER_KIND_MIIGUNNER {
```
Comment (jobrien97) [inline fighters/miigunner/src/acmd/other.rs:332]: Any reason you're checking for owner kind here? Will it ever not be Gunner?

File: fighters/miigunner/src/acmd/other.rs [hunk 2]
```diff
@@ -377,9 +431,9 @@ unsafe fn miigunner_supermissile_burst_game(fighter: &mut L2CAgentBase) {
     frame(lua_state, 1.0);
     if is_excute(fighter) {
         let gunner = utils::util::get_battle_object_from_id(owner_id);
-        if VarModule::is_flag(gunner, vars::common::instance::IS_HEAVY_ATTACK) {
-           VarModule::off_flag(gunner, vars::common::instance::IS_HEAVY_ATTACK);
-            ATTACK(fighter, 0, 0, Hash40::new("top"), 20.0, 50, 80, 0, 70, 14.0, 0.0, 0.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_ON, *ATTACK_LR_CHECK_F, false, 0, 0.0, 0, true, true, false, false, false, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_LL, *COLLISION_SOUND_ATTR_MAGIC, *ATTACK_REGION_BOMB);
+        if VarModule::is_flag(gunner, vars::miigunner::status::MISSILE_DETONATE) {
```
Comment (jobrien97) [inline fighters/miigunner/src/acmd/other.rs:434]: No need for this block, as status variables reset on status change

**Review by jobrien97 — APPROVED** [review]
lgtm

**Pull request merged**